### ISSUE TITLE
Engine crash in AI code

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#617]
```
Experienced crash, log of the last turn:

24244332: FreeActionsList called
24244332: [Cmd] /nextturn
24244332: [IPC out] N
24244332: Next turn: time 24244332
24244332: Wind = 0..343007363_1006314
24244383: Think thread started
24244880: [IPC out] +
24245596: [Cmd] /setweap ?
24245596: [IPC out] w
24245796: [Cmd] +left
24245796: [IPC out] L
24245797: [Cmd] -left
24245797: [IPC out] l
24246127: [Cmd] +down
24246127: [IPC out] D
24246604: [Cmd] -down
24246604: [IPC out] d
24247346: [Cmd] +attack
24247346: /+attack: hedgehog's Gear^.State = 2
24247346: [IPC out] A
24248277: [Cmd] -attack
24248277: [IPC out] a
24248277: AddGear: #4022 (2757,1609), 
d(-0..182028738_-2069302900,-0..428132033_1472371497) type = gtShell
24248277: [IPC out] a
24248319: [IPC out] #
24248319: [IPC out] #
24248319: Think thread started
```

Original issue reported on code.google.com by `unC0Rr` on 26 Apr 2013 at 5:43

**Comment by GoogleCodeExporter** [comment]
```
Looks very suspicious. Usually it would look like

34209791: [IPC out] #
34209791: hiTicks increment message sent

but instead there are two # messages and no notice.
Could be related to desync in long AI games.
```

Original comment by `unC0Rr` on 26 Apr 2013 at 6:01

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision c321264439c6.
```

Original comment by `unC0Rr` on 26 Apr 2013 at 8:10
* Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in rc321264439c6
```

Original comment by `unC0Rr` on 26 Apr 2013 at 8:10
